platform/api/esx/api.go

```diff
@@ -282,6 +282,34 @@ func (a *API) GetConsoleOutput(name string) (string, error) {
 	return string(buf), nil
 }
 
+func (a *API) CleanupDevice(name string) error {
+	defaults, err := a.getServerDefaults()
+	if err != nil {
+		return fmt.Errorf("couldn't get server defaults: %v", err)
+	}
+
+	_, err = defaults.finder.VirtualMachine(a.ctx, name)
+	if err == nil {
+		return fmt.Errorf("VM still exists")
+	}
+
+	// Remove the serial.out file
+	fm := defaults.datastore.NewFileManager(defaults.datacenter, true)
+	uri := fmt.Sprintf("%s/serial.out", name)
+	err = fm.DeleteFile(a.ctx, uri)
+	if err != nil {
+		return fmt.Errorf("couldn't delete serial.out: %v", err)
```

Comment:
As discussed OOB, the way to check files involves using a `Stat` method on the datastore object which we'd like to avoid.

Pushing an update to add an additional check before we print the error message that `was not found` is the suffix for the error string (they don't return an `os.PathError` but rather a string).